test/evergreen.yml

```diff
@@ -2487,6 +2484,7 @@ tasks:
           script: |
             set -o errexit
             set -o verbose
+            ${PREPARE_TEST_ENV}
```

Comment:
A few tests were missing the `TSAN_OPTIONS` environment variable, causing them to throw TSan errors in the No Throw variant